client/src/main/java/org/apache/rocketmq/client/consumer/DefaultMQPushConsumer.java

```diff
@@ -166,11 +166,16 @@
     private int consumeConcurrentlyMaxSpan = 2000;
 
     /**
-     * Flow control threshold
+     * Flow control threshold on queue level
      */
     private int pullThresholdForQueue = 1000;
 
     /**
+     * Flow control threshold on topic level, should be greater or equals than pullThresholdForQueue
+     */
+    private int pullThresholdForTopic = 2000;
```

Comment:
Then the topic flow control will happen first, so no threadhold of queue will be reached